community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java

```diff
@@ -54,57 +72,158 @@
  */
 public class Config implements DiagnosticsProvider, Configuration
 {
-    private final Map<String, String> params = new ConcurrentHashMap<>();
-    private final Iterable<Class<?>> settingsClasses;
+    private final List<ConfigOptions> configOptions;
+
+    private final Map<String,String> params = new ConcurrentHashMap<>();
     private final ConfigurationMigrator migrator;
-    private final ConfigurationValidator validator;
     private final Optional<File> configFile;
-
-    private ConfigValues settingsFunction;
-
+    private final List<ConfigurationValidator> validators = new ArrayList<>();
     // Messages to this log get replayed into a real logger once logging has been instantiated.
-    private final BufferingLog bufferedLog = new BufferingLog();
-    private Log log = bufferedLog;
+    private Log log;
 
+    /**
+     * @return a configuration with embedded defaults
+     */
     public static Config empty()
     {
-        return new Config();
+        return embeddedDefaults( Optional.empty() );
     }
 
+    /**
+     * @return a configuration with embedded defaults
+     */
     public static Config defaults()
     {
-        return new Config();
+        return embeddedDefaults( Optional.empty() );
+    }
+
+    /**
+     * @return a configuration with embedded defaults
+     */
+    public static Config embeddedDefaults( Map<String,String> additionalConfig )
+    {
+        return embeddedDefaults( Optional.empty(), additionalConfig );
+    }
```

Comment:
It certainly would but I'm not sure what values that would entail except pagecache size? I'd be happy to take suggestions on the matter